src/nsmgr/internal/pkg/flags/flags.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2020 Cisco and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package flags
+
+import (
+	"fmt"
+	"net/url"
+	"path"
+	"strings"
+
+	"github.com/networkservicemesh/cmd-nsmgr/src/nsmgr/internal/pkg/constants"
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+
+	"github.com/networkservicemesh/cmd-nsmgr/src/nsmgr/internal/pkg/values"
+)
+
+const (
+	envPrefix = "NSM"
+)
+
+// DefinedFlags - a set of flag values
+type DefinedFlags struct {
+	BaseDir string
+
+	SpiffeAgentURL url.URL
+	RegistryURL    url.URL
+
+	// Some environment variables
+	Insecure bool
+	Name     string
+
+	DeviceAPIListenEndpoint string
+	DeviceAPIRegistryServer string
+	DeviceAPIPluginPath     string
+}
+
+// Defaults - default values loaded from environment
+var Defaults = &DefinedFlags{
+	Insecure:                true,
```

Comment:
Please don't reintroduce the Insecure flag... with [spiffeutils.NewTLSPeer](https://github.com/networkservicemesh/sdk/blob/master/pkg/tools/spiffeutils/tls_peer.go#L109) you will either get a SpiffeId from Spire if its available or a self signed one... there is no reason to be running without TLS.